### ISSUE TITLE
No input validations when submit on toolbar custom actions 

### ISSUE THREAD

**Issue by MickaelMuller** [marmelab/react-admin#4295]
I have a problem after migrating to v3 on a custom action on the `Toolbar`.

At the validation of my form, I have to make two calls to my API.
 
Before the migration I used the `handleSubmit `props on a button to get the form data and make these 2 calls.
In V3 I can't use the `handleSubmit `of react-final-form, I'm using the `useFormState `to retrieve the data. 

The problem is that in v2 I had the validations on each input who have the `validate` prop but in v3 I can submit the form with empty inputs, I don't have the validations anymore.

here my Toolbar code in v2 : 

```
class Toolbar extends Component {
  render() {
    const {
      customSaveButton,
      handleSubmit,
      handleSubmitWithRedirect,
      onSubmit,
      record,
      resource,
      saveButton,
      translate,
    } = this.props;

    return (
      <Root>
        <StyledDiv>
          {customSaveButton && (
            <SaveButton
              id="SLM_saveButton"
              label={translate('ui_save')}
              onClick={handleSubmit(data => onSubmit(data))}
              submitOnEnter
            />
          )}
          {saveButton && (
            <SaveButton
              handleSubmitWithRedirect={handleSubmitWithRedirect}
              id="SLM_saveButton"
              label={isNilOrEmpty(record) ? translate('ui_save') : translate('ui_update')}
              resource={resource}
              submitOnEnter
            />
          )}
        </StyledDiv>
      </Root>
    );
  }
}

export default Toolbar;
```

In v3 : 
```
const Toolbar = ({
  customSaveButton,
  handleSubmitWithRedirect,
  onSubmit,
  record,
  resource,
  saveButton,
  saving,
  translate,
  invalid,
}) => {
  const formState = useFormState();

  const handleCustomSubmit = useCallback(() => {
    onSubmit(formState.values, invalid);
  }, [formState.values]);

  return (
    <Root>
      <StyledDiv>
        {customSaveButton && (
          <Button id="SLM_saveButton" onClick={handleCustomSubmit}>
            <Save />
            {translate(customSaveButton ? 'ui_save' : 'ui_update')}
          </Button>
        )}
        {saveButton && (
          <SaveButton
            handleSubmitWithRedirect={handleSubmitWithRedirect}
            id="SLM_saveButton"
            label={translate(isNilOrEmpty(record) ? 'ui_save' : 'ui_update')}
            resource={resource}
            saving={saving}
            submitOnEnter
            variant="text"
          />
        )}
      </StyledDiv>
    </Root>
  );
};
```

And here the component who call api, the `CreateBaseCampaign ` contain inputs with `validate `prop :

```
class CampaignCreate extends Component {
  onSubmit = (data, invalid) => {
    const {
      createCampaign,
      createLineItem,
      history: { push },
    } = this.props;
    const { levers } = data;

    if (!invalid) {
      createCampaign(
        `${CAMPAIGNS}`,
        {
          ...data,
        },
        {},
        false,
        ({
          payload: {
            data: { id },
          },
        }) =>
          createLineItem(
            `${CAMPAIGNS}/${id}/lineitems`,
            levers,
            {},
            { text: 'campaign_created', level: 'info' },
            () => push(`/${CAMPAIGNS}/${id}`),
          ),
      );
    }
  };

  render() {
    return (
      <Create {...this.props}>
        <SimpleForm
          toolbar={
            <Toolbar
              url={`/${CAMPAIGNS}`}
              customSaveButton
              saveButton={false}
              onSubmit={this.onSubmit}
            />
          }
          redirect={null}
        >
          <CreateBaseCampaign {...this.props} />
        </SimpleForm>
      </Create>
    );
   }
 }

 export default CampaignCreate;
```

**Environment**

* React-admin version: v3.1.2
* Last version that did not exhibit the issue (if applicable): 2.8
* React version: 16.12.0
* Browser:
* Stack trace (in case of a JS error):

**Comment by djhi** [comment]
Thanks for reporting this. If you are able to illustrate the bug or feature request with an example, please provide a sample application via one of the following means:

* CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple)
* A sample application via GitHub

**Comment by Madebymaurice** [comment]
Hi,

We made a [CodeSandBox](https://wwkkg.csb.app/#/posts/create).
When you load the page, if you click on the "Save" button, the form is not submitted, but the required input are not highlight in red as before.

Thanks for your help

**Comment by Madebymaurice** [comment]
I think, this not work because the meta `submitError` from react-final-form is not transmitted in your input. Actually if we use the `SimpleForm` from react-admin and we try to submit with empty required input it's the meta `error` which is passed to true.

**Comment by djhi** [comment]
Reproduced. Thanks for the codesandbox!

**Comment by jdemangeon** [comment]
Seems to be fixed by #4458 :) Closing the issue
